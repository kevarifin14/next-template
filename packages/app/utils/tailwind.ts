export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};
