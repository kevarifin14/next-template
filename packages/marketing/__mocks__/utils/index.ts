import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
