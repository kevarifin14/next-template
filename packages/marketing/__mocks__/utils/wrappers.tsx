import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
