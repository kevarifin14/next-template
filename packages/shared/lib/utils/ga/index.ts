export const pageview = (url) => {
  window.gtag("config", process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS, {
    page_path: url,
  });
};

type EventCategory =
  | "lead"
  | "outbound link"
  | "call to action"
  | "login"
  | "swap";

export const event = (
  category: EventCategory,
  action,
  label = "",
  params = {}
) => {
  if (process.env.NODE_ENV === "production") {
    window.gtag("event", action, {
      event_category: category,
      event_label: label,
      ...params,
    });
  }
};
