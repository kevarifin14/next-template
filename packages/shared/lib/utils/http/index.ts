import QueryString from 'query-string';

export function generateQueryString(params) {
  return QueryString.stringify(params);
}

export function generateEndpoint(endpoint, params) {
  if (Object.keys(params).length === 0) {
    return endpoint;
  }
  return `${endpoint}?${generateQueryString(params)}`;
}

export const get = async (endpoint, params = {}) => {
  const endpointWithParams = generateEndpoint(endpoint, params);
  const response = await fetch(endpointWithParams);

  if (response.status === 200) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};

export const post = async (endpoint, body, headers?) => {
  const response = await fetch(
    endpoint,
    {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'Content-Type': 'application/json',
        ...headers,
      },
    },
  );

  if (response.status >= 200 && response.status < 400) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};
