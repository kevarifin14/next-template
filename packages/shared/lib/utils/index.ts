export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
