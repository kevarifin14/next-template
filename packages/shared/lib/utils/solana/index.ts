/* istanbul ignore file */
import {
  AccountLayout,
  u64,
  AccountInfo as TokenAccountInfo,
  MintInfo,
} from "@solana/spl-token";
import { PublicKey } from "@solana/web3.js";

export const deserializeTokenAccount = (address: PublicKey, data: Buffer) => {
  const {
    mint,
    owner,
    amount,
    delegateOption,
    delegate,
    delegatedAmount,
    state,
    isNative,
    isNativeOption,
    closeAuthority,
    closeAuthorityOption,
  } = AccountLayout.decode(data);

  const accountInfo: TokenAccountInfo = {
    address,
    mint: new PublicKey(mint),
    owner: new PublicKey(owner),
    amount: u64.fromBuffer(amount),
    isInitialized: state !== 0,
    isFrozen: state === 2,

    ...(delegateOption === 0
      ? { delegate: null, delegatedAmount: new u64(0) }
      : {
          delegate: new PublicKey(delegate),
          delegatedAmount: u64.fromBuffer(delegatedAmount),
        }),

    ...(isNativeOption === 1
      ? { rentExemptReserve: u64.fromBuffer(isNative), isNative: true }
      : { rentExemptReserve: null, isNative: false }),

    ...(closeAuthorityOption === 0
      ? { closeAuthority: null }
      : { closeAuthority: new PublicKey(closeAuthority) }),
  };

  return accountInfo;
};

export const diffTokenAccountInfoAmounts = (
  prevTokenAccountInfo: TokenAccountInfo,
  tokenAccountInfo: TokenAccountInfo
) => {
  if (!prevTokenAccountInfo) {
    return tokenAccountInfo.amount;
  }

  return tokenAccountInfo.amount.sub(prevTokenAccountInfo.amount);
};

export const fromLamports = (
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo
) => {
  if (!tokenAccountInfo) {
    return 0;
  }

  const amount =
    typeof tokenAccountInfo === "number"
      ? tokenAccountInfo
      : tokenAccountInfo.amount.toNumber();

  const precision = 10 ** (mintInfo?.decimals || 9);
  return amount / precision;
};

export function formatTokenAmount(
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo,
  prefix = "",
  suffix = "",
  precision = 3
): string {
  return `${[prefix]}${fromLamports(tokenAccountInfo, mintInfo).toFixed(
    precision
  )}${suffix}`;
}

export const formatDollar = (value) => {
  const formatter = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
  });

  return formatter.format(value);
};
