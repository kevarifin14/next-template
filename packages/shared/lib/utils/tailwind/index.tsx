import { ComponentProps } from 'react';

export type TailwindSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl';

export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};

export const sizeToClassName = (size: TailwindSize) => {
  const dimension = sizeToDimension[size];
  return `h-${dimension} w-${dimension}`;
};

export type AlertType = 'success' | 'info' | 'error' | 'warning';

export type Heroicon = (props: ComponentProps<'svg'>) => JSX.Element; // eslint-disable-line