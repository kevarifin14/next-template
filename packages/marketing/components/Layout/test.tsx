import { render } from '@testing-library/react';

import { appearanceProviderWrapper } from '__mocks__/utils/wrappers';

import { getLayout, Layout } from '.';

describe('Layout', () => {
  const renderLayout = (props = {}) => (
    render(
      <Layout
        {...props}
      >
        hi
      </Layout>,
      { wrapper: appearanceProviderWrapper },
    )
  );

  it('matches snapshot', () => {
    const { baseElement } = renderLayout();
    expect(baseElement).toMatchSnapshot();
  });

  it('getLayout', () => {
    const { baseElement } = render(getLayout('test'));
    expect(baseElement).toMatchSnapshot();
  });
});
