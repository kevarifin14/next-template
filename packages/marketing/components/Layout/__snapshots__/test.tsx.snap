// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout getLayout 1`] = `
<body>
  <div>
    <main>
      <div
        class="min-h-screen flex flex-col"
      >
        <div
          class="px-4 sm:px-6 lg:px-8 py-2"
        >
          <div
            class="mx-auto relative flex items-center justify-between h-16 max-w-7xl w-full"
          >
            <a
              href="/"
            >
              <img
                alt="logo"
                class="mx-auto w-auto h-8 sm:h-10"
                src="/logo.png"
              />
            </a>
            <div
              class="hidden md:flex justify-center space-x-8 flex-1"
            >
              <a
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                href="/packages"
              >
                Packages
              </a>
              <a
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                href="/docs"
              >
                Docs
              </a>
              <a
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                href="/guides"
              >
                Guides
              </a>
            </div>
            <div
              class="hidden md:flex items-center justify-end space-x-8"
            >
              <button
                class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
              >
                <svg
                  class="h-6 w-6"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
                  />
                </svg>
              </button>
            </div>
            <div>
              <div
                class="md:hidden"
              >
                <button
                  class="prose dark:prose-light dark:hover:bg-dark-light hover:bg-light-dark rounded-md p-2 inline-flex items-center justify-center focus:outline-none"
                  type="button"
                >
                  <svg
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        test
      </div>
      <footer
        class="bg-dark"
      >
        <div
          class="mx-auto max-w-md py-12 px-4 overflow-hidden sm:max-w-3xl sm:px-6 lg:max-w-7xl lg:px-8"
        >
          <nav
            aria-label="Footer"
            class="-mx-5 -my-2 flex flex-wrap justify-center"
          >
            <div
              class="px-5 py-2"
            >
              <a
                class="text-base text-gray-400 hover:text-gray-300"
                href="/privacy-policy"
              >
                Privacy Policy
              </a>
            </div>
            <div
              class="px-5 py-2"
            >
              <a
                class="text-base text-gray-400 hover:text-gray-300"
                href="/terms-and-conditions"
              >
                Terms & Conditions
              </a>
            </div>
          </nav>
          <div
            class="mt-8 flex justify-center space-x-6"
          >
            <a
              class="text-gray-400 hover:text-gray-300"
              href="https://twitter.com/intent/follow?screen_name=notionql"
            >
              <span
                class="sr-only"
              >
                Twitter
              </span>
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </a>
            <a
              class="text-gray-400 hover:text-gray-300"
              href="mailto:kevin@notionql.com"
            >
              <span
                class="sr-only"
              >
                Mail
              </span>
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
                />
                <path
                  d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
                />
              </svg>
            </a>
          </div>
          <p
            class="mt-8 text-center text-base text-gray-400"
          >
            Â© 2021 App, Inc. All rights reserved.
          </p>
        </div>
      </footer>
    </main>
  </div>
</body>
`;

exports[`Layout matches snapshot 1`] = `
<body>
  <div>
    <div
      class=""
    >
      <div
        class="transform transition-colors h-full bg-light dark:bg-dark"
      >
        <main>
          <div
            class="min-h-screen flex flex-col"
          >
            <div
              class="px-4 sm:px-6 lg:px-8 py-2"
            >
              <div
                class="mx-auto relative flex items-center justify-between h-16 max-w-7xl w-full"
              >
                <a
                  href="/"
                >
                  <img
                    alt="logo"
                    class="mx-auto w-auto h-8 sm:h-10"
                    src="/logo.png"
                  />
                </a>
                <div
                  class="hidden md:flex justify-center space-x-8 flex-1"
                >
                  <a
                    class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                    href="/packages"
                  >
                    Packages
                  </a>
                  <a
                    class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                    href="/docs"
                  >
                    Docs
                  </a>
                  <a
                    class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                    href="/guides"
                  >
                    Guides
                  </a>
                </div>
                <div
                  class="hidden md:flex items-center justify-end space-x-8"
                >
                  <button
                    class="text-gray-500 hover:text-gray-900 dark:text-white dark:hover:text-primary no-underline font-medium"
                  >
                    <svg
                      class="h-6 w-6"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
                      />
                    </svg>
                  </button>
                </div>
                <div>
                  <div
                    class="md:hidden"
                  >
                    <button
                      class="prose dark:prose-light dark:hover:bg-dark-light hover:bg-light-dark rounded-md p-2 inline-flex items-center justify-center focus:outline-none"
                      type="button"
                    >
                      <svg
                        class="h-6 w-6"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 20 20"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            hi
          </div>
          <footer
            class="bg-dark"
          >
            <div
              class="mx-auto max-w-md py-12 px-4 overflow-hidden sm:max-w-3xl sm:px-6 lg:max-w-7xl lg:px-8"
            >
              <nav
                aria-label="Footer"
                class="-mx-5 -my-2 flex flex-wrap justify-center"
              >
                <div
                  class="px-5 py-2"
                >
                  <a
                    class="text-base text-gray-400 hover:text-gray-300"
                    href="/privacy-policy"
                  >
                    Privacy Policy
                  </a>
                </div>
                <div
                  class="px-5 py-2"
                >
                  <a
                    class="text-base text-gray-400 hover:text-gray-300"
                    href="/terms-and-conditions"
                  >
                    Terms & Conditions
                  </a>
                </div>
              </nav>
              <div
                class="mt-8 flex justify-center space-x-6"
              >
                <a
                  class="text-gray-400 hover:text-gray-300"
                  href="https://twitter.com/intent/follow?screen_name=notionql"
                >
                  <span
                    class="sr-only"
                  >
                    Twitter
                  </span>
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                    />
                  </svg>
                </a>
                <a
                  class="text-gray-400 hover:text-gray-300"
                  href="mailto:kevin@notionql.com"
                >
                  <span
                    class="sr-only"
                  >
                    Mail
                  </span>
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
                    />
                    <path
                      d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
                    />
                  </svg>
                </a>
              </div>
              <p
                class="mt-8 text-center text-base text-gray-400"
              >
                Â© 2021 App, Inc. All rights reserved.
              </p>
            </div>
          </footer>
        </main>
      </div>
    </div>
  </div>
</body>
`;
